spec/calendar.html

```diff
@@ -804,191 +794,200 @@ <h1>Temporal.Calendar.prototype.dateUntil ( _one_, _two_, _options_ )</h1>
-      <h1>Temporal.Calendar.prototype.daysInMonth ( _dateOrDateTime_ )</h1>
+      <h1>Temporal.Calendar.prototype.daysInMonth ( _temporalDateLike_ )</h1>
       <p>
         An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the `daysInMonth` method as specified in the ECMA-402 specification.
         If an ECMAScript implementation does not include the ECMA-402 API the following specification of the `daysInMonth` method is used.
       </p>
       <p>
-        The `daysInMonth` method takes one argument _dateOrDateTime_.
+        The `daysInMonth` method takes one argument _temporalDateLike_.
         The following steps are taken:
       </p>
       <emu-alg>
         1. Let _calendar_ be the *this* value.
         1. Perform ? RequireInternalSlot(_calendar_, [[InitializedTemporalCalendar]]).
         1. Assert: _calendar_.[[Identifier]] is *"iso8601"*.
-        1. If _dateOrDateTime_ does not have [[ISOYear]] and [[ISOMonth]] internal slots, then
-          1. Set _dateOrDateTime_ to ? ToTemporalDate(_dateOrDateTime_).
-        1. Return 𝔽(! ISODaysInMonth(_dateOrDateTime_.[[ISOYear]], _dateOrDateTime_.[[ISOMonth]])).
+        1. If Type(_temporalDateLike_) is not Object or _temporalDateLike_ does not have an [[InitializedTemporalDate]] or [[InitializedTemporalYearMonth]] internal slots, then
+          1. Set _temporalDateLike_ to ? ToTemporalDate(_temporalDateLike_).
+        1. Return 𝔽(! ISODaysInMonth(_temporalDateLike_.[[ISOYear]], _temporalDateLike_.[[ISOMonth]])).
       </emu-alg>
     </emu-clause>
 
     <emu-clause id="sec-temporal.calendar.prototype.daysinyear">
-      <h1>Temporal.Calendar.prototype.daysInYear ( _dateOrDateTime_ )</h1>
+      <h1>Temporal.Calendar.prototype.daysInYear ( _temporalDateLike_ )</h1>
       <p>
         An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the `daysInYear` method as specified in the ECMA-402 specification.
         If an ECMAScript implementation does not include the ECMA-402 API the following specification of the `daysInYear` method is used.
       </p>
       <p>
-        The `daysInYear` method takes one argument _dateOrDateTime_.
+        The `daysInYear` method takes one argument _temporalDateLike_.
         The following steps are taken:
       </p>
       <emu-alg>
         1. Let _calendar_ be the *this* value.
         1. Perform ? RequireInternalSlot(_calendar_, [[InitializedTemporalCalendar]]).
         1. Assert: _calendar_.[[Identifier]] is *"iso8601"*.
-        1. Let _year_ be ? ISOYear(_dateOrDateTime_).
-        1. Return 𝔽(! ISODaysInYear(_year_)).
+        1. If Type(_temporalDateLike_) is not Object or _temporalDateLike_ does not have an [[InitializedTemporalDate]] or [[InitializedTemporalYearMonth]] internal slot, then
+          1. Set _temporalDateLike_ to ? ToTemporalDate(_temporalDateLike_).
+        1. Return 𝔽(! ISODaysInYear(_temporalDateLike_.[[ISOYear]])).
```

Comment:
altho in this case, the F() and ISODaysInYear() kind of seem cleaner as-is, so maybe all my previous suggestions should be ignored?

Reply:
This was the reason why I left them all that way, though I realize the other places you marked above are not ideal. I'll leave this for now but if you change your mind I don't mind changing it either.

Reply:
We’ll see what the editors say.